File: app_codebase/python_base_05/bin/core/errors/error_spec.py
"""ErrorSpec — metadata for core and module error codes."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class ErrorSpec:
    code: str
    message: str
    http_status: int
    fatal_ws: bool = False

    def is_module_code(self) -> bool:
        return "/" in self.code

File: app_codebase/dart_bkend_base_02/bin/core/errors/error_spec.dart
/// Metadata for core and module error codes.
library;

class ErrorSpec {
  const ErrorSpec(
    this.code,
    this.message,
    this.httpStatus, {
    this.fatalWs = false,
  });

  final String code;
  final String message;
  final int httpStatus;
  final bool fatalWs;

  bool get isModuleCode => code.contains('/');
}
